### ISSUE TITLE
Duda sobre uso de View en metodos de otras clases

### ISSUE THREAD

**Issue by v4rgas** [RodrigoToroIcarte/IIC2113-2023-1#163]
Estoy entre dos opciones pero no sabria cual es la más correcta

Todo esto ocurre dentro de una clase que controla los turnos

```C#
 CardSet cardSet = _view.AskUserWhatSetOfCardsHeWantsToSee();
 _currentPlayer.ShowCards(cardSet, _view);
```

vs

```C#
 _currentPlayer.ShowCards(_view);
```

En la primera opción, la ventaja es que elimina la necesidad de que Player maneje el input. 

Sin embargo, dado que tanto los métodos para output como para input son parte de la clase 'View', sería suficiente pasar la clase completa como argumento, permitiendo que 'Player' la utilice dentro del método

Creo que ambas son bien parecidas pero no se si segun clean code hay una mejor

**Comment by v4rgas** [comment]
creo que otra opcion podria ser

```C#
CardSet cardSet = _view.AskUserWhatSetOfCardsHeWantsToSee();
List<string> cards = _currentPlayer.GetCardSet(cardSet);
_view.ShowCards(cards);
```

asi player solo hace consultas sobre lo que tiene y no usa View

**Comment by RodrigoToroIcarte** [comment]
La última opción me parece mejor.

**Comment by v4rgas** [comment]
Y si player tuviese como atributo view?

Seria
```
_currentPlayer.ShowCards(cardSet)
```
igual de valido? o es preferible el enfoque de la ultima opción?

tambien me causa duda para el caso de recibir daño, quizas es bueno que player tenga view porque se podria hacer un player.ReceiveDamage(int damage) y se podría revelar menos de como funciona el player internamente a cambio de que la clase Player tenga parte de la logica del juego dentro

**Comment by RodrigoToroIcarte** [comment]
Sí, es una buena solución (al menos para la E2). En la E3 se te puede complicar un poco la cosa cuando se agreguen reversals. Pero es parte del proceso de desarrollar cualquier software medianamente complejo. Mi consejo es que hagas un buen diseño en la E2. Y luego rediseñes lo que tengas que rediseñar en la E3. Igual no será tanto trabajo. Si al final resulta que tener el ShowCards o el ReceiveDamage en Player no te está funcionando, es cosa de mover ese código a otro lado :)

**Comment by v4rgas** [comment]
muchas gracias
